mqtt/handler_test.go

```diff
@@ -0,0 +1,391 @@
+package mqtt_test
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/mqtt"
+	"github.com/mainflux/mainflux/mqtt/mocks"
+	"github.com/mainflux/mainflux/pkg/errors"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/uuid"
+	"github.com/mainflux/mproxy/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	thingID        = "thingID"
+	chanID         = "123e4567-e89b-12d3-a456-000000000001"
+	invalidChanID  = "1"
+	clientID       = "clientID"
+	invalidThingID = "invalidThingID"
+	password       = "password"
+)
+
+var (
+	invalidTopic  = "invalidTopic"
+	idProvider    = uuid.NewMock()
+	logBuffer     = bytes.Buffer{}
```

Comment:
Add a comment, something like:
Test log messages for cases the handler does not provide a return value.